### DISCUSSION TITLE
How to abort execution of DynamicOutput list if one fails

### DISCUSSION THREAD

**Discussion by lbstoller** [dagster-io/dagster#12081] [Q&A] [answered]
Hi. Say I have a list of DynamicOutputs that I call .map() on. Then one of the them throws a Failure(). Is it possible to abort the execution of the ones in the list that have not yet started to execute? 

Thanks

**Comment by jamiedemaria** [comment, marked as answer]
Hey @lbstoller - there isn't a native way to do this, but you can write it yourself by querying the step events within the dynamic op

```
@op(out=DynamicOut())
def get_inputs(context):
    outputs = range(1, 10)
    for output in outputs:
        yield DynamicOutput(output, mapping_key=str(output))

@op
def my_dynamic_op(context, i: int) -> None:
    instance = context.instance
    failed_steps = instance.get_records_for_run(context.run_id, of_type=DagsterEventType.STEP_FAILURE)

    if len(failed_steps.records) > 0:
         raise Exception("failed because another step failed)

    # mock out a failure for testing
    if i == 3:
        raise Exception("This op failed")

@job
def my_job():
    data_input = get_inputs()
    results = data_input.map(my_dynamic_op)
```

**Reply by lbstoller** [reply]
Thanks! I will give this a try today.

**Reply by jamiedemaria** [reply]
great! let me know if you run into any issues - the code snippet may have a syntax error or two since i wrote directly in here - we can update it accordingly once it's working!

**Reply by lbstoller** [reply]
Hi. get_records_for_run returns:

```EventLogConnection(records=[], cursor='eyJ0eXBlIjogIlNUT1JBR0VfSUQiLCAidmFsdWUiOiAtMX0=', has_more=False)```

and len() on that returns non-zero. Is len() the correct way to get at that?

**Reply by jamiedemaria** [reply]
it should be `len(failed_steps.records)` which should get the `records` list in the `EventLogConnection` object. And i believe that should be correct

**Reply by lbstoller** [reply]
Ah sorry, about that. I should have used copy/paste. :-)

**Reply by lbstoller** [reply]
So with that fixed, this does indeed work. Thank you very much!
